app/models/maestrano/connector/rails/synchronization.rb

```diff
@@ -41,7 +41,7 @@ def set_partial
     def clean_synchronizations
       count = self.organization.synchronizations.count
       if count > 100
-        self.organization.synchronizations.limit(count - 100).destroy_all
+        self.organization.synchronizations.order('id ASC').limit(count - 100).destroy_all
```

Comment:
Skeptical on this one 😞
I've saw and replicated the issue on prod (when creating a 101st sync, it was deleting the first synchronization instead of the last one), but I've failed to create a spec to reproduce it. Crossing my finger that that will fix it.